src/java/org/apache/cassandra/db/compaction/BackgroundCompactionRunner.java

```diff
@@ -443,9 +446,15 @@ public static void handleCompactionError(Throwable t, ColumnFamilyStore cfs)
     {
         t = Throwables.unwrapped(t);
 
+        // in case of no-space-left exception, check with disk error handler in JVMStabilityInspector
+        if (Throwables.isCausedBy(t, IOException.class) && t.getMessage().contains(NO_SPACE_LEFT_MESSAGE))
```

Comment:
t.getMessage() may return null, I suggest to use:
t.toString().contains(NO_SPACE_LEFT_MESSAGE)

Reply:
do you have a full stacktrae of the error ? why we are not wrapping it with FSDiskFullWriteError ? 
I think that it should be better to have here FSDiskFullWriteError

Reply:
> do you have a full stacktrae of the error ? 

https://github.com/riptano/cndb/issues/10859#issuecomment-2359196692

> why we are not wrapping it with FSDiskFullWriteError ?

`FSDiskFullWriteError` is considered recoverable and the compactor throws `FSDiskFullWriteError` when checking disk space before starting flush or compaction task.

`IOException` with no-space-left is thrown by JDK/OS that it can't handle the write. We should consider it non-recoverable given those corruption issues.

Reply:
> t.toString().contains(NO_SPACE_LEFT_MESSAGE)

updated